components/webgpu/lib.rs

```diff
@@ -296,20 +314,69 @@ impl WGPU {
             receiver,
             sender,
             script_sender,
-            global: wgpu::hub::Global::new("wgpu-core", factory),
+            global: wgpu::hub::Global::new("wgpu-core", factory, wgt::BackendBit::PRIMARY),
             adapters: Vec::new(),
             devices: Vec::new(),
             _invalid_adapters: Vec::new(),
+            buffer_maps: HashMap::new(),
             webrender_api: webrender_api_sender.create_api(),
             webrender_document,
             external_images,
             wgpu_image_map,
         }
     }
 
-    fn run(mut self) {
+    fn run(&'a mut self) {
         while let Ok(msg) = self.receiver.recv() {
+            self.devices.iter().for_each(|device| {
```

Comment:
we can use [poll_all_devices](https://github.com/gfx-rs/wgpu/blob/29e286bdc1474457c2a34155d559e30268df74ad/wgpu-core/src/device/mod.rs#L2733) instead